pkg/pipeline/ingest/ingest_grpc.go

```diff
@@ -109,41 +91,23 @@ func instrumentGRPC(port int, counter func(int)) grpc2.UnaryServerInterceptor {
 		// instrument difference between flow time and ingest time
 		for _, entry := range flowRecords.Entries {
 			delay := timeReceived.Sub(entry.TimeFlowEnd.AsTime()).Seconds()
-			processDelaySummary.Observe(delay)
+			m.latency.Observe(delay)
 		}
 
-		// instruments number of decoded flow messages
-		flowDecoderCount.With(
-			prometheus.Labels{"worker": "", "name": decoderName}).Inc()
-
-		// instruments number of processed individual flows
-		counter(len(flowRecords.Entries))
-
-		// extract sender IP address
-		remoteIP := "unknown"
-		if md, ok := metadata.FromIncomingContext(ctx); ok {
-			if auth := md.Get(":authority"); len(auth) > 0 {
-				if portIdx := strings.IndexByte(auth[0], ':'); portIdx > 0 {
-					remoteIP = auth[0][:portIdx]
-				} else {
-					remoteIP = auth[0]
-				}
-			}
-		}
-		trafficLabels := prometheus.Labels{
-			"type":       "protobuf",
-			"remote_ip":  remoteIP,
-			"local_ip":   "0.0.0.0",
-			"local_port": localPort,
-		}
+		// instrument batch size distribution (which also instruments total flows counter under the hood)
+		m.batchSize.Observe(float64(len(flowRecords.Entries)))
```

Comment:
For information, I removed a couple of labels that I don't think are super useful, and it simplifies the code & process (no need to extract a remote IP from metadata)

Let me know if you think we should keep these labels (type, remote_ip,local_ip, local_port)